enhancements/insights/pulling-data-from-ocm.md

```diff
@@ -0,0 +1,133 @@
+---
+title: pulling-and-exposing-data-from-ocm
+authors:
+  - "@tremes"
+reviewers:
+  - "@sbose78"
+  - "@inecas"
+  - "@petli-openshift"
+  - "@smarterclayton"
+approvers:
+  - "@sbose78"
+  - "@smarterclayton"
+creation-date: 2021-03-04
+last-updated: 2021-03-09
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Insights Operator pulling and exposing data from the OCM API
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement will enable the Insights Operator to pull the data (SCA certs)
+from the OCM (OpenShift Cluster Manager) API. The data will be exposed by the Insights Operator
+in the OpenShift API to allow users to use them when consuming and building container images
+on the platform.
+
+## Motivation
+
+Users could consume RHEL content and container images using the RHEL subscription in the OpenShift 3.x.
+In the OpenShift 4, this is no longer possible because the Red Hat Enterprise Linux Core OS (RHCOS) does not
+provide any attached subscription. This enhancement is to provide users the Simple Content Access (SCA) certs
+from Red Hat Subscription Manager (RHSM).
+
+### Goals
+
+- Extend the Insights Operator config with an OCM API URL to be able to query the data
+- Periodically pull the data from the OCM API and expose it in the OpenShift API
+
+### Non-Goals
+
+- Insights Operator providing any transformation or post-processing of the data pulled
+  from the OCM API
+
+## Proposal
+
+### User Stories
+
+#### Consume SCA certs exposed in the API
+
+As an OpenShift user
+I want to consume SCA certs to be able to consume RHEL content and to build
+corresponding container images.
+
+### Risks and Mitigations
+
+#### OCM API is down
+
+Risk: OCM API is down or doesn't provide up to date data.
+
+Risk: Insights Operator is unable to expose/update the data in the OpenShift API
+
+Mitigation: Introduce a new state in the Insights Operator (e.g "SCADataDegraded") and
+create a new alert based on this new state.
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the data from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config-managed` namespace).
```

Comment:
who put that secret there? it is managed, so some operator has to sync the secret there :-)

Reply:
A good question - is this something that the installer does, or does something like the CVO sync the pull secret in `openshift-config` to `openshift-config-managed`?

Reply:
looking at the code, it looks like it's just using the pull-secret from openshift-config directly, not openshift-config-managed.

so the expectation is that secret is maintained by someone (if it expires, your nodes will have issues pulling images), so i don't think it's the responsibility of this EP to worry about it, other than that this EP should address what happens if it can't talk to OCM because the token expires(or for any other reason), which i think the EP does already.

see: https://github.com/tremes/insights-operator/blob/154641a9fae7c5100f7dce3cf0eddf4b38d56cea/pkg/config/configobserver/configobserver.go#L78

Reply:
I guess the question was why the new `etc-pki-entitlement` secret is created in the `openshift-config-managed` namespace (rather than `openshift-config` ?). 

This requires a new `Role` & `RoleBinding` definition to be able to create the secret in the `openshift-config-managed` namespace. Is that OK/acceptable? Comparing to extending our existing role to be able to create the secret in the `openshift-config` namespace.

Reply:
> I guess the question was why the new etc-pki-entitlement secret is created in the openshift-config-managed namespace (rather than openshift-config ?).

Configuration in `openshift-config` are usually added manually by admins compared, given that `etc-pki-entitlement` is managed by an OpenShift component, Adam and I suggested it be managed in the `openshift-config-managed` namespace.

Reply:
@mfojtik Hi, are you Ok if we resolve this conversation based on the comments above?